### ISSUE TITLE
Cyclic dependency problem

### ISSUE THREAD

**Issue by sffc** [zbraniecki/tinystr#21]
In my Cargo.toml, if I put

```
// Note: Hard-code patch version 0.3.4 since that is the first version with the desired feature
tinystr = { version = "0.3.4", features = ["macros"] }
```

Then I get

```
error: cyclic package dependency: package `tinystr v0.3.4` depends on itself. Cycle:
package `tinystr v0.3.4`
    ... which is depended on by `tinystr-macros v0.1.0`
    ... which is depended on by `tinystr v0.3.4`
```

**Comment by Manishearth** [comment]
We need to remove feature=macros, or factor out any of the code tinystr-macros needs into a separate crate and use reexports

**Comment by sffc** [comment]
When will we be able to make proc macros in the same crate as library code?

**Comment by Manishearth** [comment]
Unlikely to happen anytime soon. Some people want it, but it's tricky and there are no plans.

**Comment by sffc** [comment]
I fixed this in #26.
